java-checks/src/main/java/org/sonar/java/checks/tests/NoTestInTestClassCheck.java

```diff
@@ -113,7 +115,17 @@ private static boolean isArchUnitTestClass(Symbol.TypeSymbol classSymbol) {
   }
 
   private void checkTestNGmembers(IdentifierTree className, Stream<Symbol> members) {
-    if (members.noneMatch(member -> member.isMethodSymbol() && member.isPublic() && !member.isStatic() && !"<init>".equals(member.name()))) {
+    Predicate<SymbolMetadata.AnnotationInstance> isTestNgAnnotation = ann -> {
+      Type type = ann.symbol().type();
+      return type.isUnknown() || type.is(TEST_NG_TEST);
+    };
+    Predicate<Symbol> isTestMethod = member -> {
+      boolean annotatedWithTest = member.isMethodSymbol() && member.metadata().annotations().stream().anyMatch(isTestNgAnnotation);
+      boolean publicMethod = member.isMethodSymbol() && member.isPublic() && !member.isStatic() && !"<init>".equals(member.name());
+      return annotatedWithTest || publicMethod;
```

Comment:
It looks like we are only testing on methods because we have `isMethodSymbol` in both branches. Can we return early if the member is not a method?
```suggestion
      if (!member.isMethodSymbol()) {
        return false;
      }
      boolean annotatedWithTest = member.metadata().annotations().stream().anyMatch(isTestNgAnnotation);
      boolean publicMethod = member.isPublic() && !member.isStatic() && !"<init>".equals(member.name());
      return annotatedWithTest || publicMethod;
```